src/System.Collections/src/System/Collections/Generic/Stack.cs

```diff
@@ -308,12 +308,18 @@ public T Current
             {
                 get
                 {
-                    if (_index == -2) throw new InvalidOperationException(SR.InvalidOperation_EnumNotStarted);
-                    if (_index == -1) throw new InvalidOperationException(SR.InvalidOperation_EnumEnded);
+                    if (_index < 0)
+                        ThrowEnumerationNotStartedOrEnded(); // separated into a new method so the jit can inline this
```

Comment:
Nit: I'd remove this comment.  Such separation is done in lots of places all over corefx and without being commented as such in most places.